### PR TITLE
DNN-8513: Create a new extension point in the DNN core at the Database Connection Level.

### PR DESCRIPTION
**Pull request by galatrash**
Refer to this PR #1401 for details.
Jira [DNN-8513](https://dnntracker.atlassian.net/browse/DNN-8513)
>I would like the ability to break away from using the standard SqlConnection, and switch to using ReliableSqlConnection for Azure sites. I would also like the ability to set up retry policies when performing database operations.
>
>I have/will be including a code fix that adds a new extension point at the database connection layer. We can then use a custom database connection that will allow us to use ReliableSqlConnection and the RetryManager functionality that Azure provides, without having to bring those things into the DNN Core.
>
>I have created an AzureSqlConnection Provider to test this functionality, which can be found here: https://github.com/JoshuaBradley/DnnAzureSqlConnectionProvider.

### CODE REVIEW

File: DNN Platform/DotNetNuke.Web/Common/DotNetNukeHttpApplication.cs [hunk 1]
```diff
@@ -82,7 +82,9 @@ private void Application_Start(object sender, EventArgs eventArgs)
 
             ComponentFactory.Container = new SimpleContainer();
 
+            ComponentFactory.InstallComponents(new ProviderInstaller("databaseConnection", typeof(DatabaseConnectionProvider), typeof(SqlDatabaseConnectionProvider)));
             ComponentFactory.InstallComponents(new ProviderInstaller("data", typeof(DataProvider), typeof(SqlDataProvider)));
+            ComponentFactory.InstallComponents(new ProviderInstaller("databaseConnection", typeof(DatabaseConnectionProvider), typeof(SqlDatabaseConnectionProvider)));
```
Comment (zyhfish) [inline DNN Platform/DotNetNuke.Web/Common/DotNetNukeHttpApplication.cs]: why we need add two same code here with line 85?

File: Website/development.config [hunk 2]
```diff
@@ -158,7 +159,7 @@
     <globalization culture="en-US" uiCulture="en" requestEncoding="UTF-8" responseEncoding="UTF-8" fileEncoding="UTF-8"/>
     <!--<globalization culture="en-US" uiCulture="en"  fileEncoding="iso-8859-1" requestEncoding="iso-8859-1" responseEncoding="iso-8859-1"/>-->
     <!-- page level options -->
-    <pages validateRequest="false" enableViewStateMac="true" enableEventValidation="true" viewStateEncryptionMode="Always">
+    <pages validateRequest="false" enableViewStateMac="true" enableEventValidation="true"  viewStateEncryptionMode="Always">
```
Comment (zyhfish) [inline Website/development.config]: no need duplicate space here.

**Comment by galatrash** [comment]
@mitchelsellers these obsoleted public methods are already removed in 9.2.0 since they were deprecated pre v7.0. Therefore, this addition won't have a breaking change from 9.2 to 10.0.

As for fixing the conflict, someone has to take care of that as I'm no longer affiliated with DNN Corp.

**Comment by bdukes** [comment]
@mitchelsellers, I've rebased this onto `development`, and moved the config change from 9.2.0 to 9.3.0 (I didn't see anywhere that referenced the 9.2.0 config file, does that get run automatically based on version number?).  I also reverted the `DataProvider.cs` changes which were unnecessary.

**Comment by galatrash** [comment]
The config files are copied at compile time from `Dnn.Platform\DNN Platform\Website\Install\Config` folder to `Dnn.Platform\Website\Install\Config`

**Comment by mitchelsellers** [comment]
@dnnsoftware/tag Input on this?

**Pull request merged**